### PR TITLE
Accept list of pairs for gradients in GradientDescent.

### PR DESCRIPTION
**Pull request by dwf**
Be liberal about what you accept and conservative about what you emit.

### CODE REVIEW

**Comment by dwf** [comment]
Okay, tests pass now. @dmitriy-serdyuk I think this is pretty uncontroversial.

File: blocks/algorithms/__init__.py [hunk 1]
```diff
@@ -266,6 +268,8 @@ def __init__(self, cost=None, parameters=None, step_rule=None,
         # Set initial values for cost, parameters, gradients.
         self.cost = cost
         self.parameters = parameters
+        if gradients is not None and not isinstance(gradients, Mapping):
```
Comment (dmitriy-serdyuk) [inline blocks/algorithms/__init__.py]: Do you need to check if it is `Mapping`? Can gradients just always be converted to `OrderedDict`?

Comment (dwf) [inline blocks/algorithms/__init__.py]: The reason we check if it's not Mapping is that we don't want to create a dict to an OrderedDict and give it an arbitrary order.

Comment (dwf) [inline blocks/algorithms/__init__.py]: ceeabeb clarifies this in a comment.

**Comment by dmitriy-serdyuk** [comment]
It makes sense, LGTM.

**Pull request merged**